reps/2023-04-28-remove-algorithms-from-rllib.md

```diff
@@ -0,0 +1,193 @@
+from ray.tune.registry import register_env
+
+# from ray.rllib.algorithms.maml import MAML, MAMLConfig  # BEFORE
+from rllib_maml.maml import MAML, MAMLConfig  # AFTER
+
+if __name__ == "__main__":
+    ray.init()
+    register_env(
+        "cartpole",
+        lambda env_cfg: TimeLimit(CartPoleMassEnv(), max_episode_steps=200),
+    )
+
+    config = MAMLConfig()
+
+    tuner = tune.Tuner(
+        MAML,
+        param_space=config.to_dict(),
+        run_config=air.RunConfig(stop={"training_iteration": 100}),
+    )
+    results = tuner.fit()
+
+```
+
+
+This proposed algorithms to remove are:
+- A3C
+- AphaStar
+- AlphaZero
+- ApexDQN
+- ApexDDPG
+- DDPG
+- ARS
+- Bandits 
+- CRR 
+- DDPG
+- DDPPO
+- Dreamer
+- DT
+- ES
+- Leelachess
+- MADDPG
+- MAML
+- PG
+- QMix
+- Random
+- SimpleQ
+- SlateQ
+- TD3
+- R2D2
+
+
+### Shepherd of the Proposal (should be a senior committer)
+
+@sven1977
+
+## Design and Architecture
+
+- We introduce the new  repo for deprecated RLlib algorithms and community contributed algorithms.
+
+### `rayproject/rllib-contrib` File Structure
+
+```
+README.md
+.github/
+    workflow/
+        algorithm_foo.yml <---------- This will launch the unit test for algorithm foo with gh actions
+algorithm_foo/
+    src/
+        algorithm_foo.py
+    tests/
+    examples/
+    requirements.txt
+    pyproject.toml  <---------- Used for installation
+
+latest_results/
+    algorithm_foo/
+        results.json
+        results.csv
+        results.pkl
+        results.tensorboard
+        results.md  <---------- This will contain some pictures of the performance as a sanity check
+```
+
+### Installation
+1. use git to clone `rayproject/rllib-contrib`
+2. navigate to the directory of the previously deprecated algorithm
+   for ex. `cd rllib-contrib/maml`
+3. run `pip install -e .` to install the algorithm as a pip package
+4. Import the algorithm, its policies and its config from that package instead of from `ray.rllib`
+
+
+## Contribution Guidlines
+
+
+The RLlib team commits to the following level of support for the algorithms in this repo:
+
+| Platform | Purpose | Estimated Response Time | Support Level |
+| --- | --- | --- | --- |
+| [Discuss Forum](https://discuss.ray.io) | For discussions about development and questions about usage. | < 1 week | Community |
+| [GitHub Issues](https://github.com/ray-project/rllib-contrib-maml/issues) | For reporting bugs and filing feature requests. | N/A | Community |
+| [Slack](https://forms.gle/9TSdDYUgxYs8SA9e8) | For collaborating with other Ray users. | < 2 weeks | Community |
+
+**This means that any issues that are filed will be solved best-effort by the community and there is no expectation of maintenance by the RLlib team.**
+
+We will generally accept contributions to this repo that meet any of the following criteria:
+1. Updating dependencies.
+2. Submitting community contributed algorithms that have been tested and are ready for use.
```

Comment:
Let's be more specific on the exact things that an algo should be able to do/learn. Like
a) should have a quick compilation test that checks it's capabilities to learn in obs/action spaces
b) simple learning test (CartPole, or similar)
c) hard task learning benchmark that should be pinned to a specific commit so it's 100% reproducable and doesn't get "lost" over time.

Reply:
done